### DISCUSSION TITLE
// middlewares

### DISCUSSION THREAD

**Discussion by Abhiie** [axios/axios#5869] [Q&A]
// middlewares
app.use(cors());
app.use(cookieParser());
app.use(express.json());

this is servers.js file.

const googleAuth = async (req, res, next) => {
  const { name, email, img } = req.body;
  try {
    // check if user exists or not
    const user = await User.findOne({ email });

    if (user) {
      const token = generateToken(user._id);

      res.cookie("access_token", token).status(200).json({
        success: true,
        user,
      });
    } else {
      // Create new user from Google Provider
      const newUserFromGoogle = await User.create({
        name,
        email,
        img,
        fromGoogle: true,
      });
      const token = generateToken(newUserFromGoogle._id);

      res.cookie("access_token", token).status(200).json({
        success: true,
        user: newUserFromGoogle,
      });
    }
  } catch (error) {
    console.log(error);
    res.status(400);
    return next(error);
  }
};

module.exports = {
  loginUser,
  signupUser,
  googleAuth,
};
above code is from auth.js file //controllers
const jwt = require("jsonwebtoken");

const protect = async (req, res, next) => {
  try {
    // Get token from header
    const token = req.cookies.access_token;

    if (!token) {
      res.status(401);
      return next(new Error("Not authorized, no token"));
    }

    // Verify token
    const decoded = jwt.verify(token, process.env.JWT_SECRET);

    // Get user id from token and assign to req
    req.user = { _id: decoded._id };
    next();
  } catch (error) {
    res.status(401);
    return next(new Error("Not authorized"));
  }
};

module.exports = protect;
above code is from auth.js //middlewares
but still i am not able to set cookies help me

_Originally posted by @Abhiie in https://github.com/axios/axios/issues/4907#issuecomment-1499894681_

**Comment by EdwardsVO** [comment]
Do you solve it? I have the same issue with my auth middleware

**Reply by EdwardsVO** [reply]
I solved it, the problem was in the cookie domain

**Reply by anakin-dabir** [reply]
wow
